src/Avalonia.Controls/TextBox.cs

```diff
@@ -714,7 +714,8 @@ protected override void OnPointerPressed(PointerPressedEventArgs e)
 
         protected override void OnPointerMoved(PointerEventArgs e)
         {
-            if (_presenter != null && e.Pointer.Captured == _presenter)
+            // selection should not change during pointer move if the user right clicks
+            if (_presenter != null && e.Pointer.Captured == _presenter && !e.GetCurrentPoint(this).Properties.IsRightButtonPressed)
```

Comment:
I wonder if this should be `&& e.GetCurrentPoint(this).Properties.IsLeftButtonPressed` because this still allows a middle-button drag which shouldn't work iiuc?

Reply:
You're right. I've adjusted this!